cpp/include/cudf/scalar/scalar.hpp

```diff
@@ -590,5 +592,58 @@ struct duration_scalar : chrono_scalar<T> {
    */
   typename T::rep count() { return this->value().count(); }
 };
+
+/**
+ * @brief An owning class to represent a list value in device memory
+ */
+class list_scalar : public scalar {
+ public:
+  using value_type = cudf::list_view;
+
+  list_scalar() : scalar(data_type(type_id::LIST)) {}
+  ~list_scalar()                        = default;
+  list_scalar(list_scalar&& other)      = default;
+  list_scalar(list_scalar const& other) = default;
+  list_scalar& operator=(list_scalar const& other) = delete;
+  list_scalar& operator=(list_scalar&& other) = delete;
+
+  /**
+   * @brief Construct a new list scalar object from existing device data
+   *
+   * @param value The value of the list
+   * @param is_valid Whether the value held by the scalar is valid
+   * @param stream CUDA stream used for device memory operations.
+   * @param mr Device memory resource to use for device memory allocation
+   */
+  list_scalar(cudf::column_view const& value,
+              scalar_validity validity            = scalar_validity::VALID,
+              rmm::cuda_stream_view stream        = rmm::cuda_stream_default,
+              rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource())
+    : scalar(data_type(type_id::LIST), validity == scalar_validity::VALID ? true : false),
+      _data(value, stream, mr)
+  {
+  }
+
+  /**
+   * @brief Get the value of the scalar as a list_view
+   *
+   * @param stream CUDA stream used for device memory operations.
+   */
+  value_type value(rmm::cuda_stream_view stream = rmm::cuda_stream_default) const;
```

Comment:
1. I find calling this function `value` confusing.
2. Do we need a function that returns a `list_view` *and* a function that returns a column_view? 

If we can eliminate the `column_view` returning function and just call the list_view returning function `view()` it solves both 1 and 2.

Reply:
Thanks. Previously I was not paying attention to what `list_view` really represents. But I guess the docstring made it pretty clear - `A non-owning, immutable view of device data that represents a list of elements of arbitrary type`. Which coincides the precise definition of `data`.

Reply:
While I agree with not having 2 separate functions, we might not be able to use `list_view` without modifications. It's an empty struct right now.